pkg/odo/cli/component/create_methods.go

```diff
@@ -60,9 +61,17 @@ func (icm InteractiveCreateMethod) FetchDevfileAndCreateComponent(co *CreateOpti
 		if err != nil {
 			return err
 		}
-	} else {
-		// if the user is logged in or if we have cluster information, display the default project
+	} else if co.KClient != nil {
+		// Get the current project if no --project is passed, but --now is passed
 		componentNamespace = ui.EnterDevfileComponentProject(co.KClient.GetCurrentProjectName())
+	} else {
+		var client kclient.ClientInterface
+		client, err = kclient.New()
+		// Continue if an error is encountered
+		if err == nil && client != nil {
+			// Get the current project if no --project is passed and using offline context
+			componentNamespace = ui.EnterDevfileComponentProject(client.GetCurrentProjectName())
+		}
```

Comment:
If kclient.New() fails, componentNamespace won't be set. Isn't it problematic? Also, don't we want to get it from the envfile?

Reply:
>Also, don't we want to get it from the envfile?

We will first need to obtain the (desired) namespace from the user, only then can we add it to the envfile. So, at this stage, there is no question of obtaining the namespace from envfile.